streams/jvm/src/main/scala/zio/stream/experimental/platform.scala

```diff
@@ -119,4 +125,288 @@ trait ZStreamPlatformSpecificConstructors {
   ): ZStream[R, E, A] =
     effectAsyncInterrupt(k => register(k).toRight(UIO.unit), outputBuffer)
 
+  /**
+   * Creates a stream of bytes from a file at the specified path.
+   */
+  def fromFile(path: => Path, chunkSize: Int = ZStream.DefaultChunkSize): ZStream[Any, Throwable, Byte] =
+    ZStream
+      .bracket(ZIO.effectBlockingInterrupt(FileChannel.open(path)))(chan => ZIO.effectBlocking(chan.close()).orDie)
+      .flatMap { channel =>
+        ZStream.fromEffect(UIO(ByteBuffer.allocate(chunkSize))).flatMap { reusableBuffer =>
+          ZStream.repeatEffectChunkOption(
+            for {
+              bytesRead <- ZIO.effectBlockingInterrupt(channel.read(reusableBuffer)).asSomeError
+              _         <- ZIO.fail(None).when(bytesRead == -1)
+              chunk <- UIO {
+                         reusableBuffer.flip()
+                         Chunk.fromByteBuffer(reusableBuffer)
+                       }
+            } yield chunk
+          )
+        }
+      }
+
+  /**
+   * Creates a stream from a `java.io.InputStream`.
+   * Note: the input stream will not be explicitly closed after it is exhausted.
+   */
+  def fromInputStream(
+    is: => InputStream,
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[Any, IOException, Byte] =
+    ZStream.fromEffect(UIO(is)).flatMap { capturedIs =>
+      ZStream.repeatEffectChunkOption {
+        for {
+          bufArray  <- UIO(Array.ofDim[Byte](chunkSize))
+          bytesRead <- ZIO.effectBlockingIO(capturedIs.read(bufArray)).asSomeError
+          bytes <- if (bytesRead < 0)
+                     ZIO.fail(None)
```

Comment:
Per JavaDoc
_the total number of bytes read into the buffer, or -1 if there is no more data because the end of the stream has been reached._
`bytesRead <0 `is thus expected (and not a failure) and `bytesRead==0` would never occur